src/hooks/utility/view.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2023-2024 Retake, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::ffi::CStr;
+
+use anyhow::Result;
+use pg_sys::{get_relname_relid, RangeVarGetCreationNamespace};
+
+use pgrx::*;
+
+use crate::{duckdb::connection::execute, hooks::query::is_duckdb_query};
+
+use super::set_search_path_by_pg;
+
+pub fn view_query(query_string: &core::ffi::CStr, stmt: *mut pg_sys::ViewStmt) -> Result<bool> {
+    let query = unsafe { (*stmt).query as *mut pg_sys::SelectStmt };
+    let from_clause = unsafe { (*query).fromClause };
+
+    if analyze_from_clause(from_clause)? {
+        // Push down the view creation query to DuckDB
+        set_search_path_by_pg()?;
+        execute(query_string.to_str()?, [])?;
+    }
+
+    Ok(true)
```

Comment:
Here, the original utility hook is executed after this function. Maybe we should execute the original utility hook first to exit early and report errors when there are issues with the CREATE VIEW statement.

Maybe we should refactor the code to allow the original hook to be executed earlier.